service/controller.go

```diff
@@ -315,63 +331,35 @@ func (s *service) CreateVolume(
 	}
 
 	thick := params[ThickVolumesParam]
-
-	applicationPrefix := ""
-	if params[ApplicationPrefixParam] != "" {
-		applicationPrefix = params[ApplicationPrefixParam]
-	}
+	applicationPrefix := s.resolveParameter(params, symmetrixID, ApplicationPrefixParam, "")
 
 	// Storage (resource) Pool. Validate it against exist Pools
-	storagePoolID := params[StoragePoolParam]
+	storagePoolID := s.resolveParameter(params, symmetrixID, StoragePoolParam, "")
 	err = s.validateStoragePoolID(ctx, symmetrixID, storagePoolID, pmaxClient)
 	if err != nil {
 		log.Error(err.Error())
 		return nil, status.Errorf(codes.InvalidArgument, "%s", err.Error())
 	}
 
 	// SLO is optional
-	serviceLevel := "Optimized"
-	if params[ServiceLevelParam] != "" {
-		serviceLevel = params[ServiceLevelParam]
-		found := false
-		for _, val := range validSLO {
-			if serviceLevel == val {
-				found = true
-			}
-		}
-		if !found {
-			log.Error("An invalid Service Level parameter was specified")
-			return nil, status.Errorf(codes.InvalidArgument, "An invalid Service Level parameter was specified")
+	serviceLevel := s.resolveParameter(params, symmetrixID, ServiceLevelParam, "Optimized")
+	found := false
+	for _, val := range validSLO {
+		if serviceLevel == val {
+			found = true
```

Comment:
Can we break the for loop once `found` becomes true for the first time?

Reply:
Added the break.